hazelcast/src/test/java/com/hazelcast/internal/util/collection/NonTraversableListTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util.collection;
+
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+import static java.util.function.UnaryOperator.identity;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class NonTraversableListTest {
```

Comment:
In my opinion, this `test` method API is a bit overcomplicated, and the tests using `test` method API are not explicit : it's quite hard to easily understand the logic. Would you mind to rewrite the tests in more explicit way?

Reply:
Is [this patch][1] enough, or do you prefer a fluent API, i.e. test builders?

[1]: https://github.com/hazelcast/hazelcast/files/13599725/Rename.test.methods.patch

Reply:
In my opinion, not enough. Why not to make each test explicit, even with potential duplications? Why, if you're testing `size()` not just call size with assertions in function body? And so on.

Reply:
I am against the approach where readability is sacrificed for duplication prejudices.

Reply:
>  I can switch to a fluent API

Try it, lets see if it reads better.

Reply:
7fe7c69b

Reply:
✔️